wallet/src/main/scala/org/bitcoins/wallet/models/WalletStateDescriptorDAO.scala

```diff
@@ -0,0 +1,100 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.commons.jsonmodels.wallet.WalletStateDescriptorType._
+import org.bitcoins.commons.jsonmodels.wallet.{
+  SyncHeightDescriptor,
+  WalletStateDescriptor,
+  WalletStateDescriptorType
+}
+import org.bitcoins.crypto.DoubleSha256DigestBE
+import org.bitcoins.db.{CRUD, SlickUtil}
+import org.bitcoins.wallet.config.WalletAppConfig
+import slick.lifted.ProvenShape
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class WalletStateDescriptorDb(
+    tpe: WalletStateDescriptorType,
+    descriptor: WalletStateDescriptor) {
+  require(descriptor.descriptorType == tpe)
+}
+
+case class WalletStateDescriptorDAO()(implicit
```

Comment:
So the plan is to throw them all in the same table as strings? I would prefer to have some sort of structure in the database so I can query for simple things height or block hash.

Reply:
I used strings so it can be more extensible in the future. So if we wanted to add a descriptor that had data about something that doesn't have a height and block hash, we can easily do without needing to make a new table.